src/charm.py

```diff
@@ -709,14 +709,17 @@ def _get_prefix(data: Union["RequirerData_IPU", "RequirerData_IPA"]):
     def _generate_middleware_config(
         self, data: Union["RequirerData_IPA", "RequirerData_IPU"], prefix: str
     ) -> dict:
-        """Generate a stripPrefix middleware for path based routing."""
-        if self._routing_mode is _RoutingMode.path and data.get("strip-prefix", False):
-            return {
-                f"juju-sidecar-noprefix-{prefix}": {
-                    "stripPrefix": {"prefixes": [f"/{prefix}"], "forceSlash": False}
-                }
-            }
+        """Generate a middleware config."""
+        config = {}  # type: Dict[str, Dict[str, Any]]
+        if self._routing_mode is _RoutingMode.path:
+            if data.get("strip-prefix", False):
```

Comment:
Should this (and below) be
```suggestion
            if data.get("strip-prefix") == "true":
```

Reply:
the library does the conversion
![image](https://github.com/canonical/traefik-k8s-operator/assets/6230162/a81af02a-3a29-4482-b838-ac1a4193d1d7)